Piotr Andrzejewski <piotr.andrzejewski@redembedded.com>
ONEM-38885 - interface is extended with multidecryption support
Implementation of multidecryption is not required in widevine.

diff --git a/MediaSession.cpp b/MediaSession.cpp
index 3bad443..bdb7430 100644
--- a/MediaSession.cpp
+++ b/MediaSession.cpp
@@ -435,6 +435,20 @@ CDMi_RESULT MediaKeySession::Init(
   return CDMi_SUCCESS;
 }
 
+CDMi_RESULT MediaKeySession::Decrypt(
+        uint8_t* inData,
+        const uint32_t inDataLength,
+        uint8_t** outData,
+        uint32_t* outDataLength,
+        const SampleInfo* sampleInfo,
+        const uint16_t sampleCount,
+        const IStreamProperties* properties) {
+  if (sampleCount > 1) {
+    printf("Multidecrypt not supported");
+  }
+  return Decrypt(inData, inDataLength, outData, outDataLength, sampleInfo, properties);
+}
+
 CDMi_RESULT MediaKeySession::Decrypt(
         uint8_t* inData,
         const uint32_t inDataLength,
diff --git a/MediaSession.h b/MediaSession.h
index 1df14d6..267e31c 100644
--- a/MediaSession.h
+++ b/MediaSession.h
@@ -108,6 +108,7 @@ public:
         uint8_t** outData,
         uint32_t* outDataLength,
         const SampleInfo* sampleInfo,
+        const uint16_t sampleCount,
         const IStreamProperties* properties);
 
     virtual CDMi_RESULT ReleaseClearContent(
@@ -134,6 +135,13 @@ public:
 
 private:
     void onKeyStatusError(widevine::Cdm::Status status);
+    CDMi_RESULT Decrypt(
+        uint8_t* inData,
+        const uint32_t inDataLength,
+        uint8_t** outData,
+        uint32_t* outDataLength,
+        const SampleInfo* sampleInfo,
+        const IStreamProperties* properties);
 
 private:
     widevine::Cdm *m_cdm;
